### PR TITLE
[EMB-370] User settings profile name

### PR DESCRIPTION
**Pull request by brianjgeiger**
## Purpose

Allow user to change their name parts. Note that this is based on #465 and #469. Those should probably get merged first.

## Summary of Changes

1. Added names changing components
2. Added citation preview
3. Tests

## Feature Flags

Part of `ember_user_settings_profile_page`

## QA Notes

See ticket for acceptance tests which may be used to create your tests sessions.

## Ticket

https://openscience.atlassian.net/browse/EMB-370

# Reviewer Checklist

- [ ] meets requirements
- [ ] easy to understand
- [ ] DRY
- [ ] testable and includes test(s)
- [ ] changes described in `CHANGELOG.md`

### CODE REVIEW

**Comment by brianjgeiger** [comment]
Note to self: add some analytics. Also: is there a good way to test the no transition if model is dirty?

File: app/settings/profile/name/-components/citation-preview/component.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+@tagName('')
+export default class CitationPreview extends Component {
+    // Required Parameter
+    user!: User;
+
+    @and('user.givenName', 'user.familyName')
+    shouldCite!: boolean;
+
+    initials = (names: string): string => {
```
Comment (jamescdavis) [inline app/settings/profile/name/-components/citation-preview/component.ts]: make regular class method

File: app/settings/profile/name/-components/citation-preview/component.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            .split(/\s+/)
+            .map(upperCaseName => {
+                return `${upperCaseName[0].toUpperCase()}.`;
+            })
+            .filter(initial => {
+                return initial.match(/^[a-z]/i);
+            }).join(' ');
+    }
+
+    suffix = (suffix: string): string => {
```
Comment (jamescdavis) [inline app/settings/profile/name/-components/citation-preview/component.ts]: also this one

File: app/settings/profile/name/-components/citation-preview/component.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        let returnValue = suffix;
+        const suffixLower = suffix.toLowerCase();
+        if ($.inArray(suffixLower, ['jr', 'sr']) !== -1) {
```
Comment (jamescdavis) [inline app/settings/profile/name/-components/citation-preview/component.ts]:
```suggestion
        if (['jr', 'sr'].includes(suffixLower)) {
```

File: app/settings/profile/name/-components/citation-preview/component.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+            returnValue = `${suffix}.`;
+            returnValue = suffixLower.charAt(0).toUpperCase() + suffixLower.slice(1);
+        } else if ($.inArray(suffixLower, ['ii', 'iii', 'iv', 'v']) !== -1) {
+            returnValue = suffixLower.toUpperCase();
+        }
+        return returnValue;
+    }
+
+    @computed('user.familyName', 'user.givenName', 'user.middleNames', 'user.suffix')
+    get citeApa (): string {
```
Comment (jamescdavis) [inline app/settings/profile/name/-components/citation-preview/component.ts]:
```suggestion
    get citeApa(): string {
```

File: app/settings/profile/name/-components/citation-preview/component.ts [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        let cite: string = '';
+        const currentUser = this.user;
+        if (this.shouldCite) {
```
Comment (jamescdavis) [inline app/settings/profile/name/-components/citation-preview/component.ts]: don't need

File: app/settings/profile/name/template.hbs [hunk 6]
```diff
@@ -0,0 +1,13 @@
+{{title (t 'settings.profile.name.title')}}
+<div class="panel panel-default">
+    <div class="panel-heading clearfix">
+        <h3 class="panel-title">{{t 'settings.profile.name.title'}}</h3>
+    </div>
+    {{settings/profile/name/-components/name 
+        user=this.user
+        onWillDestroy=(action this.onWillDestroy) 
+        onSave = (action this.onSave)
+        isPageDirty = this.user.hasDirtyAttributes
```
Comment (jamescdavis) [inline app/settings/profile/name/template.hbs]:
```suggestion
        isPageDirty=this.user.hasDirtyAttributes
```

**Pull request merged**